src/basics/basics.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+
+namespace basics;
+
+
+use JetBrains\PhpStorm\Pure;
+
+class basics implements BasicsInterface
+{
+    private BasicsValidator $validator;
+
+    #[Pure] public function __construct()
+    {
+        $this->validator = new BasicsValidator();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getMinuteQuarter(int $minute): string
+    {
+        $this->validator->isMinutesException($minute);
+
+        if ($minute > 45 or $minute == 0) {
```

Comment:
Please read difference between 'or' and '||' and use || and && instaed of 'or' 'and'